### ISSUE TITLE
[BUG]envinjector unable to pull docker image from azurecr using Azure AKS WorkloadIdentity

### ISSUE THREAD

**Issue by akhilm-cervello** [SparebankenVest/azure-key-vault-to-kubernetes#454]
**Components and versions**

AKS Version-1.24.3
akv2k8s-2.3.0
Helm version: 1.4.0

**Describe the bug**
When attempting to use env-injection with custom user-assigned managed identity using workload identity the env-injector fails to pull down the docker image with the Authorization failed error.

**To Reproduce**
Create an AKS Cluster with Workload Identity. Try to inject secrets into pods using AKV2K8S.

**Expected behavior**
pod to start up
env_injector init container to be added
main container to have the secret injected into env

**Logs**
Below are the error logs from demo pod.

Error from server (InternalError): error when creating "demo.yaml": Internal error occurred: failed calling webhook "pods.env-injector.admission.spv.no": failed to call webhook: an error on the server ("{\"response\":{\"uid\":\"93ed3abe-0393-4e0d-8219-17024435973e\",\"allowed\":false,\"status\":{\"metadata\":{},\"status\":\"Failure\",\"message\":\"failed to get auto cmd, error: GET https://acrname.azurecr.io/oauth2/token?scope=repository%3Apact%2Fdev%2Fapp%3Apull\\u0026service=acrname.azurecr.io: UNAUTHORIZED: authentication required, visit https://aka.ms/acr/authorization for more information.\\ncannot fetch image descriptor\\ngithub.com/SparebankenVest/azure-key-vault-to-kubernetes/pkg/docker/registry.getImageConfig\\n\\t/go/src/github.com/SparebankenVest/azure-key-vault-to-kubernetes/pkg/docker/registry/registry.go:144\\ngithub.com/SparebankenVest/azure-key-vault-to-kubernetes/pkg/docker/registry.(*Registry).GetImageConfig\\n\\t/go/src/github.com/SparebankenVest/azure-key-vault-to-kubernetes/pkg/docker/registry/registry.go:103\\nmain.getContainerCmd\\n\\t/go/src/github.com/SparebankenVest/azure-key-vault-to-kubernetes/cmd/azure-keyvault-secrets-webhook/registry.go:39\\nmain.podWebHook.mutateContainers\\n\\t/go/src/github.com/SparebankenVest/azure-key-vault-to-kubernetes/cmd/azure-keyvault-secrets-webhook/pod.go:143\\nmain.podWebHook.mutatePodSpec\\n\\t/go/src/github.com/SparebankenVest/azure-key-vault-to-kubernetes/cmd/azure-keyvault-secrets-webhook/pod.go:299\\nmain.vaultSecretsMutator\\n\\t/go/src/github.com/SparebankenVest/azure-key-vault-to-kubernetes/cmd/azure-keyvault-secrets-webhook/main.go:163\\ngithub.com/slok/kubewebhook/pkg/webhook/mutating.MutatorFunc.Mutate\\n\\t/go/pkg/mod/github.com/slok/kubewebhook@v0.11.0/pkg/webhook/mutating/mutator.go:25\\ngithub.com/slok/kubewebhook/pkg/webhook/mutating.mutationWebhook.mutatingAdmissionReview\\n\\t/go/pkg/mod/github.com/slok/kubewebhook@v0.11.0/pkg/webhook/mutating/webhook.go:128\\ngithub.com/slok/kubewebhook/pkg/webhook/mutating.mutationWebhook.Review\\n\\t/go/pkg/mod/github.com/slok/kubewebhook@v0.11.0/pkg/webhook/mutating/webhook.go:120\\ngithub.com/slok/kubewebhook/pkg/webhook/internal/instrumenting.(*Webhook).Review\\n\\t/go/pkg/mod/github.com/slok/kubewebhook@v0.11.0/pkg/webhook/internal/inst") has prevented the request from succeeding

Is it possible to add an Exception like in the case of AAD Pod Identity or is there something else we can do to get the keyvault injector working?

**Comment by 181192** [comment]
Hi @akhilm-cervello, we are working on support for workload identity in #442. I have released chart version 2.3.1 that have support for necessary labels to be set on service account and image versions 1.5.0-beta.1

```yaml
global:
  keyVaultAuth: environment-azidentity
controller:
  image:
    tag: 1.5.0-beta.1
  podLabels:
    azure.workload.identity/use: "true"
  serviceAccount:
    labels:
      azure.workload.identity/use: "true"
    annotations:
      azure.workload.identity/client-id: <managed-client-id>
env_injector:
  image:
    tag: 1.5.0-beta.1
  podLabels:
    azure.workload.identity/use: "true"
  replicaCount: 1
  envImage:
    tag: 1.5.0-beta.1
  serviceAccount:
    labels:
      azure.workload.identity/use: "true"
    annotations:
      azure.workload.identity/client-id: <managed-client-id>

```

Closing this issue, to track discussion on bugs in #442

**Comment by akhilm-cervello** [comment]
@181192 We are unable to get the issue resolved in spite of trying the above changes. Could you please let us know if the solution is already implemented or if the implementation is still in progress?
